pandapower/create.py

```diff
@@ -3898,6 +3904,10 @@ def _create_column_and_set_value(net, index, variable, column, element, dtyp=flo
         net[element].at[index, column] = variable
     elif default_for_nan and column in net[element].columns:
         net[element].at[index, column] = default_val
+    try:
+        net[element][column] = net[element][column].astype(dtyp)
```

Comment:
This could be somehow dangerous, as I would trust pandas' data handling more in most cases. What if you just add some column with a boolean value without explicitly giving the dtype here? At the current state of pandas, the new column will be handled as dtype object. This might still change in later versions, however, a user would probably expect the current pandas behavior rather than a cast into float64 as the default.